Cesium.CodeGen/Ir/BlockItems/CompoundStatement.cs

```diff
@@ -17,7 +17,9 @@ public CompoundStatement(Ast.CompoundStatement statement)
     {
     }
 
-    public bool HasDefiniteReturn => _items.Count > 0; // TODO[#90]: Better definite return analysis.
+    public bool HasDefiniteReturn =>
+        _items.Count > 0
+        && !_items.All(x =>!(x as DeclarationBlockItem)?.IsScopedIdentifierWithInitalizer?? false); // TODO[#90]: Better definite return analysis.
```

Comment:
Please, refactor messy predicate on the line 22. At first glance, it is seen that you have double negation which does not make any sense. It is clearly that you can simplify double negation of the part `!_items.All(x =>!(x as DeclarationBlockItem)?.IsScopedIdentifierWithInitalizer?? false)`

Next, consider the following possibilities to make predicate inside extension method `All` looking better:

- Replace `as` with `is` using predicate as follows: `x => x is DeclarationBlockItem decl && !decl.IsScopedIdentifierWithInitalizer`
- Using pattern matching `x => x is DeclarationBlockItem { IsScopedIdentifierWithInitalizer: false }`
- Defining method that compatible with the signature `Func<BlockItem, bool>` and passing it directly inside method `All`
- Combining all previous suggestions

Also, you have a typo: `IsScopedIdentifierWithInitalizer` -> `IsScopedIdentifierWithInitializer`